shared/appengine.go

```diff
@@ -11,16 +11,53 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
-	"os/exec"
-	"strings"
+	"os"
 	"time"
 
 	"github.com/google/go-github/v31/github"
 	"golang.org/x/oauth2"
-	"google.golang.org/appengine"
+	apps "google.golang.org/api/appengine/v1"
+
+	// TODO(#1747): Deprecate this library.
 	"google.golang.org/appengine/user"
 )
 
+// runtimeIdentity contains the identity of the current AppEngine service when
+// running on GAE, or empty when running locally.
+var runtimeIdentity struct {
+	LocationID  string
+	AppID       string
+	Service     string
+	Version     string
+	application *apps.Application
+}
+
+func init() {
+	// Env vars available on GAE:
+	// https://cloud.google.com/appengine/docs/standard/go/runtime#environment_variables
+	// Note: the "region code" part of GAE_APPLICATION is NOT location ID.
+	if proj := os.Getenv("GOOGLE_CLOUD_PROJECT"); proj != "" {
+		runtimeIdentity.AppID = proj
+		runtimeIdentity.Service = os.Getenv("GAE_SERVICE")
+		if runtimeIdentity.Service == "" {
+			panic("Missing GAE_SERVICE")
```

Comment:
Although we'll get a stack-trace anyway if we panic, can we be more specific about the problem here? I.e. `GAE_SERVICE environment variable not set` or similar.